src/averageAges.js

```diff
@@ -56,7 +84,20 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
-}
+  const child = people.filter(a => people.some(b => b.name === a.mother));
+  const childAges = child.map(a =>
+    a.born - people.find(b =>
+      b.name === a.mother).born);
+  const diff = childAges.reduce((a, b) => a + b) / child.length;
+
+  const sons = people.filter(a =>
+    people.some(b => a.mother === b.name) && a.sex === 'm');
```

Comment:
variable's names should be more informative
```suggestion
  const sons = people.filter(son =>
    people.some(person => son.mother === person.name) && son.sex === 'm');
```